drive/drive_test.go

```diff
@@ -0,0 +1,459 @@
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	fxfscore "github.com/functionland/go-fula/fxfs/core/api"
+	fxiface "github.com/functionland/go-fula/fxfs/core/iface"
+	"github.com/functionland/go-fula/fxfs/core/pfs"
+	"github.com/ipfs/go-datastore"
+	syncds "github.com/ipfs/go-datastore/sync"
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/ipfs/kubo/config"
+	"github.com/ipfs/kubo/core"
+	"github.com/ipfs/kubo/core/coreapi"
+	"github.com/ipfs/kubo/repo"
+)
+
+const testPeerID = "QmTFauExutTsy4XP6JbMFcw2Wa9645HJt2bTqL6qYDCKfe"
+
+func newFxFsCoreAPI() fxiface.CoreAPI {
+	r := &repo.Mock{
+		C: config.Config{
+			Identity: config.Identity{
+				PeerID: testPeerID, // required by offline node
+			},
+		},
+		D: syncds.MutexWrap(datastore.NewMapDatastore()),
+	}
+
+	ctx := context.Background()
+
+	node, err := core.NewNode(ctx, &core.BuildCfg{Repo: r})
+	if err != nil {
+		panic(err)
+	}
+
+	capi, err := coreapi.NewCoreAPI(node)
+	if err != nil {
+		panic(err)
+	}
+
+	fapi, err := fxfscore.NewCoreAPI(node, capi)
+	if err != nil {
+		panic(err)
+	}
+
+	return fapi
+}
+
+func newDrive(userDID string) (UserDrive, fxiface.CoreAPI) {
+	fapi := newFxFsCoreAPI()
+	ctx := context.Background()
+	ds := NewDriveStore()
+
+	ud, err := ds.ResolveCreate(userDID)
+	if err != nil {
+		panic(err)
+	}
+
+	ud.Publish(ctx, fapi)
+
+	return ud, fapi
+}
+
+func TestPublishDrive(t *testing.T) {
+	fapi := newFxFsCoreAPI()
+	ctx := context.Background()
+	testUserDID := "did:fula:resolves-to-mehdi-2"
+	ds := NewDriveStore()
+
+	ud, err := ds.ResolveCreate(testUserDID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ud.Publish(ctx, fapi)
+
+	fmt.Printf("%+v", ud)
+}
+
+func TestDriveSpace(t *testing.T) {
+	fapi := newFxFsCoreAPI()
+	ctx := context.Background()
+	testUserDID := "did:fula:resolves-to-mehdi-2"
+	ds := NewDriveStore()
+
+	ud, err := ds.ResolveCreate(testUserDID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ud.Publish(ctx, fapi)
+
+	ps, err := ud.PublicSpace(ctx, fapi)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = ps.MkDir("/data", MkDirOpts{recursive: false})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f := files.NewBytesFile([]byte("some data to test ListEntries method"))
+	_, err = ps.WriteFile("/data/test.txt", f, WriteFileOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ls, err := ps.ListEntries("/", ListEntriesOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("ls for /")
+	for entry := range ls {
+		if entry.Err != nil {
+			t.Fatal(entry.Err)
+		}
+		fmt.Printf("%+v \n", entry)
+	}
+
+	ls, err = ps.ListEntries("/data", ListEntriesOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("\n\nls for /data")
+	for entry := range ls {
+		if entry.Err != nil {
+			t.Fatal(entry.Err)
+		}
+		fmt.Printf("%+v \n", entry)
+	}
+
+	_, err = ps.MkDir("/data/summer", MkDirOpts{recursive: false})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ls, err = ps.ListEntries("/data", ListEntriesOpts{})
+	if err != nil {
```

Comment:
Using a the [testify](https://github.com/stretchr/testify) testing library might help reduce LOC in tests.

Reply:
it needs another PR. I created an issue for it [here](https://github.com/functionland/go-fula/issues/63)